types/type_map.go

```diff
@@ -99,3 +103,16 @@ func (t typeMap) Validate() error {
 	}
 	return t.value.Validate()
 }
+
+func genName(baseName string, key bool) string {
+	postfix := "value"
+	if key {
+		postfix = "key"
+	}
+	digest := md5.New()
+	n, err := digest.Write([]byte(baseName + postfix))
```

Comment:
1. It is not a MD5 hash
2. no bug, see https://github.com/gojuno/genval/blob/wip-17-fixed-complex-generation/examples/complicated/validators.go#L231
3. counter is a global var, so i disagree that it is better solution.